contracts/managers/withdraw/WithdrawManager.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.26;
+
+import {AccessManagedUpgradeable} from "../access/AccessManagedUpgradeable.sol";
+import {WithdrawManagerStorageLib} from "./WithdrawManagerStorageLib.sol";
+import {WithdrawRequest} from "./WithdrawManagerStorageLib.sol";
+
+struct WithdrawRequestInfo {
+    uint256 amount;
+    uint256 endWithdrawWindowTimestamp;
+    bool canWithdraw;
+    uint256 withdrawWindowInSeconds;
+}
+/**
+ * @title WithdrawManager
+ * @dev Manages withdrawal requests and their processing, ensuring that withdrawals are only allowed within specified windows and conditions.
+ */
+contract WithdrawManager is AccessManagedUpgradeable {
+    constructor(address accessManager_) {
+        initialize(accessManager_);
+    }
+
+    function initialize(address accessManager_) internal initializer {
+        super.__AccessManaged_init(accessManager_);
+    }
+
+    /**
+     * @notice Checks if the account can withdraw the specified amount and updates the withdraw request.
+     * @dev This function can only be executed by `plasmaVault`.
```

Comment:
can only be executed by account with role PLASMA_VAULT_ROLE, which is in fact Plasma Vault contract